code/rdb/rdbstandard.q

```diff
@@ -1,6 +1,32 @@
-// Get the relevant RDB attributes
-.proc.getattributes:{`date`tables!(.rdb.rdbpartition[],();tables[])}
-
+\d .proc
+/- Get the relevant RDB attributes
+getattributes:{
+    timecolumns:1!except[(::),gettimecolumns each tables[`.];(::)];
+    attrtable:`date`tables`procname!
+	({[timecolumns]
+	  d:exec(min;max)@\:distinct`date$first each raze value each timecolumns from timecolumns;
+	  first[d]+til 1+last deltas d}[timecolumns];
+	tables[];
+	.proc.procname);
+    if[.ds.datastripe;
+	filtermap:flip @[.j.k read1 hsym`$getenv[`KDBAPPCONFIG],"/striping.json";`$string first .ds.segmentid];
```

Comment:
is this not already read into some global variable or something? (possibly sent by TP when subscription occurs?) feels strange to read from an on-disk json when this function is called (presumably the GW will be calling this periodically to get the up-to-date attributes etc.)